### PR TITLE
Replace a few cairo 0 contracts in tests vol. 3

### PR DESCRIPTION
**Pull request by kkawula**
Relates #1334

## Introduced changes

- Removed `balance` cairo 0 contract
- Removed `balance_struct_event` cairo 0 contract
- Removed `complex_abi` cairo 0 contract

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: starknet_py/tests/e2e/block_test.py [hunk 1]
```diff
@@ -66,7 +66,9 @@ async def test_get_block_with_txs_pending(account):
 
 # TODO (#1419): Fix contract redeclaration
 @pytest.mark.asyncio
-async def test_get_block_with_txs_latest(account):
+async def test_get_block_with_txs_latest(account, map_class_hash):
+    # pylint: disable=unused-argument
+
```
Comment (kkawula) [inline starknet_py/tests/e2e/block_test.py:71]: Param `map_class_hash` was added in purpose to create a new block - test setup executes some transactions that cause block creation

File: starknet_py/tests/e2e/client/fixtures/prepare_network.py [hunk 2]
```diff
@@ -81,77 +97,79 @@ def fixture_invoke_transaction(
...
+
+    prepared_data = await prepare_net_for_tests(
+        account,
+        deploy_account_details=details,
+        transaction_hash=transaction_hash,
+        contract=deployed_balance_contract,
+        decalre_class_hash=class_hash,
+    )
+
+    net = devnet
```
[the remark this answers is not in the document]
Comment (kkawula) [reply starknet_py/tests/e2e/client/fixtures/prepare_network.py]: Fixed

File: starknet_py/tests/e2e/mock/cairo_0_abi/complex_abi_abi.json
[the remark this answers is not in the document]
Comment (kkawula) [reply starknet_py/tests/e2e/mock/cairo_0_abi/complex_abi_abi.json:1]: No, the contract was called [complex_abi](https://github.com/software-mansion/starknet.py/pull/1434/files#:~:text=starknet_py/tests/e2e/mock/contracts/complex_abi.cairo) in compilation process compiler creates two files and one of them containing abi has `abi` postfix added

[the remark this answers is not in the document]
Comment (franciszekjob) [reply starknet_py/tests/e2e/mock/cairo_0_abi/complex_abi_abi.json:1]: So maybe we can rename contract, for instance `complex_contract.cairo`? Cause the generated name `complex_abi_abi` does not look good imo.  Wdyt @kkawula ?

File: starknet_py/tests/e2e/fixtures/constants.py [hunk 3]
```diff
@@ -59,6 +59,7 @@ def _get_env_lambda(env_name):
 TYPED_DATA_DIR = MOCK_DIR / "typed_data"
 CONTRACTS_DIR = MOCK_DIR / "contracts"
 CONTRACTS_COMPILED_V0_DIR = MOCK_DIR / "contracts_compiled"
+CAIRO_0_ABI_DIR = MOCK_DIR / "cairo_0_abi"
```
Comment (franciszekjob) [inline starknet_py/tests/e2e/fixtures/constants.py:62]: nit:
```suggestion
CAIRO_0_CONTRACTS_ABI_DIR = MOCK_DIR / "cairo_0_contracts_abi"
```
Also please update further occurrences.

**Review by franciszekjob — APPROVED** [review]
LGTM

**Pull request merged**